From 087a1e23eb451d8fc33bd04b1e417b56f0d71692 Mon Sep 17 00:00:00 2001
From: Neil Williams <codehelp@debian.org>
Date: Mon, 2 Feb 2015 16:50:09 -0200
Subject: [PATCH 2/2] add mbr back to support extlinux and check for
 installation of mbr.

Signed-off-by: Antonio Terceiro <terceiro@debian.org>
---
 vmdebootstrap | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/vmdebootstrap b/vmdebootstrap
index 74627e2..67caa84 100755
--- a/vmdebootstrap
+++ b/vmdebootstrap
@@ -108,7 +108,7 @@ class VmDebootstrap(cliapp.Application):
                               'Create an apt source based on the distribution '
                               'and mirror selected.')
         self.settings.boolean(['mbr'],
-                              'Run install-mbr (no longer done by default)')
+                              'Run install-mbr (default if extlinux used)')
         self.settings.boolean(['grub'],
                               'Install and configure grub2 - disables '
                               'extlinux.')
@@ -136,7 +136,7 @@ class VmDebootstrap(cliapp.Application):
             if self.settings['image']:
                 self.create_empty_image()
                 self.partition_image()
-                if self.settings['mbr']:
+                if self.settings['mbr'] or self.settings['extlinux']:
                     self.install_mbr()
                 (rootdev, bootdev) = self.setup_kpartx()
                 self.mkfs(rootdev, type=roottype)
@@ -263,6 +263,11 @@ class VmDebootstrap(cliapp.Application):
         if os.path.exists("/sbin/install-mbr"):
             self.message('Installing MBR')
             self.runcmd(['install-mbr', self.settings['image']])
+        else:
+            msg = "mbr enabled but /sbin/install-mbr not found"
+            " - please install the mbr package."
+            raise cliapp.AppException(msg)
+
 
     def setup_kpartx(self):
         out = self.runcmd(['kpartx', '-avs', self.settings['image']])
-- 
2.1.4

