From: Neil Williams <codehelp@debian.org>
Date: Sun 15 Sep 20:47:37 BST 2013
Subject: Add missing commits between the 0.1.0 orig tarball
 and the actual 0.1.0 upstream tag.
Author: Neil Williams <codehelp@debian.org>
Bug-Debian: http://bugs.debian.org/721668

---

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- vmdebootstrap-0.1.0.orig/vmdebootstrap
+++ vmdebootstrap-0.1.0/vmdebootstrap
@@ -73,6 +73,11 @@ class VmDebootstrap(cliapp.Application):
                                   metavar='USER/PASSWORD')
         self.settings.boolean(['serial-console'], 
                               'configure image to use a serial console')
+        self.settings.string(['serial-console-command'],
+                             'command to manage the serial console, appended '
+                               'to /etc/inittab (%default)',
+                             metavar='COMMAND',
+                             default='/sbin/getty -L ttyS0 115200 vt100')
         self.settings.boolean(['sudo'], 
                               'install sudo, and if user is created, add them '
                                 'to sudo group')
@@ -102,6 +107,7 @@ class VmDebootstrap(cliapp.Application):
             self.set_hostname(rootdir)
             self.create_fstab(rootdir)
             self.install_debs(rootdir)
+            self.cleanup_apt_cache(rootdir)
             self.set_root_password(rootdir)
             self.create_users(rootdir)
             self.remove_udev_persistent_rules(rootdir)
@@ -109,6 +115,7 @@ class VmDebootstrap(cliapp.Application):
             self.customize(rootdir)
             if self.settings['image']:
                 self.install_extlinux(rootdev, rootdir)
+                self.optimize_image(rootdir)
             if self.settings['tarball']:
                 self.create_tarball(rootdir)
         except BaseException, e:
@@ -191,10 +198,10 @@ class VmDebootstrap(cliapp.Application):
 
         if not self.settings['no-kernel']:
             if self.settings['arch'] == 'i386':
-                kernel_arch = '686'
+                kernel_arch = '486'
             else:
                 kernel_arch = self.settings['arch']
-            kernel_image = 'linux-image-2.6-%s' % kernel_arch
+            kernel_image = 'linux-image-%s' % kernel_arch
             include.append(kernel_image)
 
         if self.settings['sudo'] and 'sudo' not in include:
@@ -246,6 +253,10 @@ class VmDebootstrap(cliapp.Application):
         logging.debug('stdout:\n%s' % out)
         shutil.rmtree(tmp)
 
+    def cleanup_apt_cache(self, rootdir):
+        out = self.runcmd(['chroot', rootdir, 'apt-get', 'clean'])
+        logging.debug('stdout:\n%s' % out)
+
     def set_root_password(self, rootdir):
         if self.settings['root-password']:
             self.message('Setting root password')
@@ -299,7 +310,7 @@ class VmDebootstrap(cliapp.Application):
         
         if self.settings['enable-dhcp']:
             f.write('\n')
-            f.write('allow-hotplug eth0\n')
+            f.write('auto eth0\n')
             f.write('iface eth0 inet dhcp\n')
             
         f.close()
@@ -345,15 +356,25 @@ append initrd=%(initrd)s root=UUID=%(uui
         f.close()
         
         if self.settings['serial-console']:
+            serial_command = self.settings['serial-console-command']
             logging.debug('adding getty to serial console')
             inittab = os.path.join(rootdir, 'etc/inittab')
             with open(inittab, 'a') as f:
-                f.write('\nS0:23:respawn:/sbin/getty -L ttyS0 115200 vt100\n')
+                f.write('\nS0:23:respawn:%s\n' % serial_command)
 
         self.runcmd(['extlinux', '--install', rootdir])
         self.runcmd(['sync'])
         import time; time.sleep(2)
-        
+
+    def optimize_image(self, rootdir):
+        """
+        Filing up the image with zeros will increase its compression rate
+        """
+        zeros = os.path.join(rootdir, 'ZEROS')
+        self.runcmd_unchecked(['dd', 'if=/dev/zero', 'of=' + zeros, 'bs=1M'])
+        self.runcmd(['rm', '-f', zeros])
+
+
     def cleanup_system(self):
         # Clean up after any errors.
 
